### PR TITLE
New plugin: You Snooze, You Lose!

### PR DESCRIPTION
**Pull request by davidryan59**
This plugin has an input for a player address or Twitter handle, it then queries their transaction history, and outputs the numbers of moves they have done in each hour of the last 24 hours. Each band is colour coded: green is not many moves (or zero), red is a lot of moves.

It is then possible to spot the time of day they are sleeping (green), and have a good idea about when they are next asleep and it is safe to whack their planets!

Voilà! You snooze – you lose!

Credit to @Bulmenisaurus for the original idea "AccountActivity.js"

### CODE REVIEW

**Review by Stx69 — DISMISSED** [review]
Hi reviewed & tested and work well for me.

**Review by Stx69 — APPROVED** [review]

**Comment by davidryan59** [comment]
This plugin requires data from The Graph

Graph API URL worked for Round 4, but I don't think it worked for Round 5, but I don't know why?

**Comment by ichub** [comment]
@Stx69 - thank you for reviewing this plugin. I was wondering if you would be interested in helping to review plugins via a more formalized process with the Dark Forest core team and some other community members? If so, please add me as a friend on Discord! ichub#7930

@davidryan59, same goes for you! Thank you for writing these plugins. We're working through the review backlog right now, would you be interested in helping out?

**Comment by ichub** [comment]
Screen Shot 2022-03-31 at 3 42 09 PM

I just tested this in production just now by making a move, and the count was updated correctly. Very cool!

**Comment by ichub** [comment]
Screen Shot 2022-03-31 at 3 43 20 PM

one thing i notice is the count moves is the same here, which makes me suspicious that moves are being double-counted somehow

**Comment by davidryan59** [comment]
Could the 21, 21 be a coincidence here?

Try again half an hour later?

The code is partitioning the moves into 24 buckets of 1 hour each, I don't believe it is possible for the same move to be counted in 2 buckets.

**Comment by davidryan59** [comment]
Screenshot 2022-04-01 at 14 41 41

I have just tested the plugin in https://zkga.me/play/0x5da117b8ab8b739346f5edc166789e5afb1a7145 and it worked nicely, see screenshot

File: content/utilities/you-snooze-you-lose/plugin.js [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+    // Results table
+    container.appendChild(this.resultsTable);
+  }
+
+  destroy() {
+  }
+}
+
+export default YouSnoozeYouLose;
```
Comment (cristobal) [inline content/utilities/you-snooze-you-lose/plugin.js:278]:
```suggestion
export default YouSnoozeYouLose;

```
nitpick Github wants a newline ;)

File: content/utilities/you-snooze-you-lose/index.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+title: You Snooze, You Lose!
+date: 2022-01-06T12:00:00+07:00
+subtitle: Find out when players are asleep, and punish them severely!
+version: 0.6.4
```
Comment (cristobal) [inline content/utilities/you-snooze-you-lose/index.md:5]:
```suggestion
version: 0.6.5
```

Bump to version `0.6.5`

**Comment by davidryan59** [comment]
From @ichub via Discord: "We could add `df.getGraphApiUrl()`. I can get that in tomorrow"

**Comment by ichub** [comment]
> From @ichub via Discord: "We could add `df.getGraphApiUrl()`. I can get that in tomorrow"

done

**Comment by davidryan59** [comment]
@ichub I have used THEGRAPH_API_URL import now on this plugin.

I've tested it v0.6.5 and it works. Ready for review?

**Review by ichub — APPROVED** [review]

**Pull request merged**